ruth/globalview.py

```diff
@@ -53,8 +53,13 @@ def level_of_service_for_car(self, datetime, segment, vehicle: Vehicle, toleranc
 
         n_vehicles = self.number_of_vehicles_in_time_at_segment(datetime, segment.id, tolerance, vehicle)
 
+        ending_length = 100
+        rest_segment_length = segment.length - vehicle.start_distance_offset
```

Comment:
The number of vehicles per mile calculated below is higher the closer the vehicle is to the end of the segment (if there are vehicles in front of it). This causes lags at the ends of the segments and also on short segments. Setting different calculation for the last 100 (now 200) meters is a temporary solution. We will probably use multiple scales for the LoS as a long-term solution instead of this one. Should we add this functionality to this PR or is it better to change it back and keep the old behaviour for now?

Also, `number_of_vehicles_in_time_at_segment` should return only the number of vehicles that are in the rest of the segment in this case.

Reply:
> Also, number_of_vehicles_in_time_at_segment should return only the number of vehicles that are in the rest of the segment in this case.

Ok, this was the missing piece. This is highly non-obvious from the code. When `vehicle_id` and `vehicle_offset_m` are used, the name of the function `number_of_vehicles_in_time_at_segment` isn't really accurate. I would just create another function for this, because it's quite confusing.